### PR TITLE
Chop game background

### LINKED ISSUE #9125: Subgame background image is stretched
The custom background image for a subgame is always stretched to the window's rectangle, breaking the aspect ratio if the window's aspect does not match the aspect of the image. This makes the feature useless for anything other than simple gradient backgrounds. Since actual artwork looks silly when deformed in this way, I'm submitting this as a bug.
A more correct way to display the image would be to uniformly scale it to touch the window from the outside.

### PR DESCRIPTION
**Pull request by appgurueu**
Closes #9125 - chops at right / bottom, ready for review.

### CODE REVIEW

**Comment by hecktest** [comment]
I made it stick to the center too, check your pulls.

**Comment by appgurueu** [comment]
> I made it stick to the center too, check your pulls.

Merged.

File: src/gui/guiEngine.cpp [hunk 1]
```diff
@@ -435,9 +435,21 @@ void GUIEngine::drawBackground(video::IVideoDriver *driver)
 		return;
 	}
 
+	v2u32 screensize_orig = screensize;
+	v2f32 scale(
+			(f32) screensize.X / sourcesize.X,
+			(f32) screensize.Y / sourcesize.Y);
+	if (scale.X < scale.Y)
+		screensize.X = (int) (scale.Y * sourcesize.X);
```
Comment (rubenwardy) [inline src/gui/guiEngine.cpp:443]: better to write to a variable called `bg_size` or similar than to write to `screensize`.

**Review by SmallJoker — APPROVED** [review]

**Review by sfan5 — APPROVED** [review]
LGTM

**Pull request merged**